securedrop_client/gui/widgets.py

```diff
@@ -377,6 +381,31 @@ def clear_message(self) -> None:
         self._hide()
 
 
+class UserGuideLabel(QLabel):
+    """ A widget that contains the user guide for journalists.
+    This opens in a Tor disposable VM.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        # Set css id
+        self.setObjectName("UserGuideLink")
+
+        self.setFixedSize(QSize(120, 22))
```

Comment:
This comment is a little bit more subtle, please bear with me :slightly_smiling_face: 

Specifying _fixed_ sizes for widgets carries the risk of making them less re-usable by coupling them to a use case. Because of that I think it would be best to avoid it where possible. Let me explain what I mean by coupling a widget to a specific use case.

The purpose of creating a widget is to group together a set of characteristics (or example, the link text, its target URL and the fact that it is clickable), in order to ensure consistency across multiple uses. (For example, this widget would make easy to ensure that every link to the user guide says `User Guide` and points to the same URL.)

The _size_ of a widget is somewhat different because very often it depends not only on what the widget itself _is_, but also in what _context_ it is used. For example, if this link was standing by itself we'd probably want to set wider margins around it than if it was, for example, part of a longer piece of text. Does this make sense?

With that in mind, it is very difficult to predict what the exact size of a widget should be. Certainly, we can say that it shouldn't be _smaller_ than a given size, but requiring it to be exactly of any given size will make it inadequate for most contexts that are different of the specific one we've in mind.
 
How can we avoid that and make sure that the widget is as re-usable as practical?
Most of the time it helps to:

- avoid fixed sizes
- let the context of the widget (the "parent" of the widget in the hierarchy of widgets) decide of its size. This means that, for example, the `UserGuideLink` **class** wouldn't dictate a size, but whoever creates a new **instance** can set a size that is appropriate to the context in which it is used.
- focus on setting minimums, for example minimum margins, or padding where necessary (often it's not necessary)

Example:

```py
class LeftPane(QWidget):
    """..."""
    def __init__(self) -> None:
        super().__init__()
        # ...

        self.user_guide_label = UserGuideLabel()
        self.user_guide_label.setFixedSize(QSize(120, 22))  # if that's the appropriate size when used in the LeftPane

        # ...
```

Another, shorter way to say this would be to say that deciding for the _size_ of a `UserGuideLink` is not part of the responsibilities of that class, it is the responsibility of whatever code _uses_ it. This is what explains why, if you're curious and start reading how some of the Qt widgets' documentation, you'll find that it is recommended that [each widget provides it's own `sizeHint`][qt] (merely suggestions to the parent), but rarely strict `size` requirements.

  [qt]: https://doc.qt.io/qtforpython/PySide6/QtWidgets/QWidget.html#size-hints-and-size-policies

Does this make sense?